registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/main/java/io/mosip/registration/processor/stages/packetclassifier/tagging/impl/MetaInfoTagGenerator.java

```diff
@@ -0,0 +1,160 @@
+package io.mosip.registration.processor.stages.packetclassifier.tagging.impl;
+
+import io.mosip.kernel.core.exception.BaseCheckedException;
+import io.mosip.registration.processor.core.constant.JsonConstant;
+import io.mosip.registration.processor.core.exception.util.PlatformErrorMessages;
+import io.mosip.registration.processor.core.packet.dto.FieldValue;
+import io.mosip.registration.processor.packet.storage.exception.ParsingException;
+import io.mosip.registration.processor.stages.packetclassifier.dto.FieldDTO;
+import io.mosip.registration.processor.stages.packetclassifier.tagging.TagGenerator;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnExpression(value = "'${mosip.regproc.packet.classifier.tag-generators}'.contains('MosipMetaInfo')")
+public class MetaInfoTagGenerator implements TagGenerator {
+
+    /** The tag name that will be prefixed with every metainfo operationsData tags */
+    @Value("${mosip.regproc.packet.classifier.tagging.metainfo.operationsdata.tag-name-prefix:META_INFO-OPERATIONS_DATA-}")
+    private String operationsDataTagNamePrefix;
+
+    /** The tag name that will be prefixed with every metainfo metaData tags */
+    @Value("${mosip.regproc.packet.classifier.tagging.metainfo.metadata.tag-name-prefix:META_INFO-META_DATA-}")
+    private String metaDataTagNamePrefix;
+
+    /** The tag name that will be prefixed with every metainfo capturedRegisteredDevices tags */
+    @Value("${mosip.regproc.packet.classifier.tagging.metainfo.capturedregistereddevices.tag-name-prefix:META_INFO-CAPTURED_REGISTERED_DEVICES-SERIAL_NO-}")
+    private String capturedRegisteredDevicesTagNamePrefix;
+
+    /** The labels on metainfo.operationsData array that needs to be tagged */
+    @Value("#{'${mosip.regproc.packet.classifier.tagging.metainfo.operationsdata.tag-labels:}'.split(',')}")
```

Comment:
no defaults will force to have a default value in config can we avoid having any configuration without default.

Reply:
Here default is given as empty string so if they don't want to tag anything in operationsdata, then they can remove the config and it will work without that